docs/dev/backport.md

```diff
@@ -36,3 +36,53 @@ keep the label
 
 5.6. Ask someone else (from QA, or the patch author, or anyone relevant)
 for assistance in selecting among the options
+
+## How to backport a patch
+
+Here are the instructions on how to backport some changes to an older version of Scylla.  
+It's the easiest to show it by an example, so let's look at PR [#12031](https://github.com/scylladb/scylladb/pull/12031), which fixed issue [#12014](https://github.com/scylladb/scylladb/issues/12014).
+
+### Wait for the original change to get merged
+First the original PR should get merged.
+In the example the author wanted to merge the branch `cvybhu:filter_multi_bug` into `scylladb:master`.
+Each PR is submitted on its own branch, which the PR author wants to merge into the `master` branch.
+Before the changes end up on `master`, they are first added to the `next` branch to run some additional tests. Later the `next` branch becomes `master` and the PR is officialy merged.
+
+### A backport PR
+Released versions of `Scylla` live on their own branches - `branch-5.0`, `branch-4.6` etc.
+To backport the change we will have to put it on its own branch and merge it into the branch with the desired version.
+In the example, the PR with the backport [#12086](https://github.com/scylladb/scylladb/pull/12086) wants to merge `cvybhu:filter_multi_bug_5.0` into `scylladb:branch-5.0`.
+Before the changes end up on `branch-5.0`, they are first added to the `next-5.0` branch to run some additional tests. Later the `next-5.0` branch becomes `branch-5.0` and the PR is officially backported.
+
+### Preparing the backport PR
+1. Go to the branch with the target backport version
+```bash
+git fetch origin branch-5.0
+git checkout -t origin/branch-5.0
+git submodule sync
+git submodule update --recursive
```

Comment:
You might need to do another `git pull origin branch-5.0` before this, as the first fetch might fail to get all the submodule changes (as it is executed before `git submodule sync`).

Reply:
Damn git is terrible.

Added `git pull` before `submodule update`

Reply:
In the usual workflow you already have `branch-5.0` and you just check it out, then do the sync and update and pull.

Reply:
I'm no git expert, but my thinking was:
* use git fetch to fetch all commits from `branch-5.0` to the local repo
* use `git checkout -t origin/branch-5.0` to create a branch matching the remote one and checkout to it
* Now the local and version controlled submodule references could apparently be out of sync so I need to do `submodule sync`
* Then use `submodule update` to actually change the files in submodule directories to the right ones

I'm still not sure why the `git pull` is needed

Reply:
I run `git submodule foreach git fetch --all` to get commits for all submodules for all repos for which I have remotes.